packages/x-components/src/x-modules/tagging/service/__tests__/pdpd-add-to-cart.service.spec.ts

```diff
@@ -0,0 +1,124 @@
+import { InMemoryStorageService } from '@empathyco/x-storage-service';
+import { Store } from 'vuex';
+import { DefaultPDPAddToCartService } from '../pdp-add-to-cart.service';
+import { XPlugin } from '../../../../plugins/index';
+import { installNewXPlugin } from '../../../../__tests__/utils';
+import { taggingXModule } from '../../x-module';
+import { RootXStoreState } from '../../../../store/index';
+import { createResultStub } from '../../../../__stubs__/index';
+
+interface PDPdAddToCartServiceTestAPI {
+  service: DefaultPDPAddToCartService;
+  localStorageService: InMemoryStorageService;
+  sessionStorageService: InMemoryStorageService;
+  store: Store<RootXStoreState>;
+}
+
+function preparePDPAddToCartService(): PDPdAddToCartServiceTestAPI {
+  XPlugin.resetInstance();
+  installNewXPlugin();
+  XPlugin.registerXModule(taggingXModule);
+
+  const store = XPlugin.store;
+  const localStorageService = new InMemoryStorageService();
+  const sessionStorageService = new InMemoryStorageService();
+  const service = new DefaultPDPAddToCartService(localStorageService, sessionStorageService);
+
+  return {
+    service,
+    localStorageService,
+    sessionStorageService,
+    store
+  };
+}
+
+describe('testing pdp add to cart', () => {
```

Comment:
Would it make sense to test that the TTL is working and the storage is cleaned after the time passes?

Reply:
IMO I don't see the reason for that, since that's responsibility of the storage service and we already have tests for that, [see browser storage service tests](https://github.com/empathyco/x/blob/main/packages/storage-service/src/__tests__/browser-storage-service.spec.ts#L45)

Reply:
Okay true. What about the case we discussed in `packages/x-components/src/x-installer/api/api.types.ts` about no `productId` when calling `addToCart`?